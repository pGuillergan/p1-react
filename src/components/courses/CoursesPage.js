import React, { useState, useEffect } from "react";
import CourseList from "./CourseList";
import { BrowserRouter, Link } from 'react-router-dom';
import Navbar from './../common/Navbar'
import Container from "react-bootstrap/Container";

export default function CoursesPage() {
  const [courses, setCourses] = useState([]);

  useEffect(() => {
    fetch("http://54.215.227.33:3000/getallcourses")
      .then((response) => response.json())
      .then((data) => {
        setCourses(data);
      })
      .catch((err) => console.log(err));
  }, []);

  return (
    <div>
      <Navbar />
      <Container>
      <h3>Courses Page</h3>
      <br />
      <Link to="/course/add">
        <button className="btn btn-primary">Add Course</button>
      </Link>

      {'  |  '}

      <Link to="/course/edit">
        <button className="btn btn-primary">Edit Course</button>
      </Link>

      {'  |  '}

      <Link to="/course/delete">
        <button className="btn btn-primary">Delete Course</button>
      </Link>
      <br />
      <br />
      <CourseList courses={courses} />
      </Container>
    </div>
  );
}
