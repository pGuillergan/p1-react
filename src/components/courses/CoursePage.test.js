import React, { Component } from "react";
import { shallow } from "enzyme";

import CoursesPage from "./CoursesPage";

describe("CoursesPage", () => {
  it("should display the list of courses", () => {
    const coursesPage = shallow(<CoursesPage/>);
    expect(coursesPage).toMatchSnapshot();

    expect(coursesPage.find('.btn').length).toEqual(3);

  });

});
