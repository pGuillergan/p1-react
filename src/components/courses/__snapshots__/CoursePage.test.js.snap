// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoursesPage should display the list of courses 1`] = `
<div>
  <Navbar />
  <br />
  <h3>
    Courses Page
  </h3>
  <br />
  <Link
    to="/course/add"
  >
    <button
      className="btn btn-primary"
    >
      Add Course
    </button>
  </Link>
    |  
  <Link
    to="/course/edit"
  >
    <button
      className="btn btn-primary"
    >
      Edit Course
    </button>
  </Link>
    |  
  <Link
    to="/course/delete"
  >
    <button
      className="btn btn-primary"
    >
      Delete Course
    </button>
  </Link>
  <br />
  <br />
  <CourseList
    courses={Array []}
  />
</div>
`;
